topdown/eq.go

```diff
@@ -32,6 +32,8 @@ func evalEqGround(t *Topdown, a ast.Value, b ast.Value, iter Iterator) error {
 	}
 	if ast.Compare(a, b) == 0 {
 		return iter(t)
+	} else if errGroundNotEqual {
+		return fmt.Errorf("scalars did not unify: %d and %d", a, b)
```

Comment:
This error is ignored by everything but the function evaluation. Without it, the following is ambiguous:
```
f(1) = y {
y = 2
}

f(2) = y {
y = 4
}
```
A call to f(1) will unify with both of those declaration, which is incorrect.